catalog/spatial/geocoding/spatial-geocoding-plugin/src/main/java/org/codice/ddf/spatial/geocoding/plugin/GeoCoderPlugin.java

```diff
@@ -118,11 +102,27 @@ public int getRadius() {
         return radiusInKm;
     }
 
-    private boolean countryCodeNeeded(Metacard metacard) {
-        return metacard.getAttribute(Location.COUNTRY_CODE) == null;
+    private void setCountryCodeAttributeIfNeeded(Metacard metacard) {
+        Optional<String> wktLocation = Optional.ofNullable(metacard.getLocation());
+
+        if (wktLocation.isPresent() && countryCodeNeeded(metacard)) {
+            GeoCoder geocoder = geoCoderFactory.getService();
+
+            if (geocoder != null) {
+                Optional<String> alpha3CountryCode = geocoder.getCountryCode(wktLocation.get(), radiusInKm);
```

Comment:
I think this would work too:

`geocoder.getCountryCode(wktLocation.get(), radiusInKm).ifPresent(countryCode -> {

  metacard.setAttribute(new AttributeImpl(Location.COUNTRY_CODE, alpha3CountryCode.get()));

                    LOGGER.trace("Setting metacard country code to {} for metacard with id {}",
                            alpha3CountryCode.get(),
                            metacard.getId());
})`